tls/extensions/s2n_key_share.c

```diff
@@ -17,19 +17,27 @@
 #include "tls/s2n_tls.h"
 #include "utils/s2n_safety.h"
 
+S2N_RESULT s2n_ecdhe_send_public_key(struct s2n_stuffer *out, struct s2n_ecc_evp_params *ecc_evp_params)
+{
+    RESULT_ENSURE_REF(ecc_evp_params);
+    RESULT_ENSURE_REF(ecc_evp_params->negotiated_curve);
+
+    RESULT_GUARD_POSIX(s2n_stuffer_write_uint16(out, ecc_evp_params->negotiated_curve->share_size));
+    if (ecc_evp_params->evp_pkey == NULL) {
+        RESULT_GUARD_POSIX(s2n_ecc_evp_generate_ephemeral_key(ecc_evp_params));
+    }
```

Comment:
Not that I can think of. The only time we generate a share twice is during a retry, and if the curve hasn't changed then there's no reason to change our share either. Shares are wiped both after they're used to calculate the secret and when a connection is wiped.

Reply:
(This also isn't a behavior change-- I added it this null check as part of the early data feature, since early data can trigger a retry and retries with the same curve are now possible)